extensions/cornerstone-dicom-sr/src/viewports/OHIFCornerstoneSRViewport.tsx

```diff
@@ -430,19 +437,76 @@ OHIFCornerstoneSRViewport.defaultProps = {
   customProps: {},
 };
 
+const findInstance = (measurement, displaySetService: DisplaySetService) => {
+  const {
+    displaySetInstanceUID,
+    ReferencedSOPInstanceUID: sopUid,
+  } = measurement;
+  const referencedDisplaySet = displaySetService.getDisplaySetByUID(
+    displaySetInstanceUID
+  );
+  if (!referencedDisplaySet.images) return;
+  return referencedDisplaySet.images.find(it => it.SOPInstanceUID === sopUid);
+};
+
 async function _getViewportReferencedDisplaySetData(
   displaySet,
   measurementSelected,
   displaySetService
 ) {
   const { measurements } = displaySet;
-  const measurement = measurements[measurementSelected];
 
-  const { displaySetInstanceUID } = measurement;
+  if (!displaySet.keyImageDisplaySet) {
+    const findReferences = ds => {
```

Comment:
Does this findReferences need to be defined inside the `_getViewportReferencedDisplaySetData` ? or worse inside the if statement?